compiler/rustc_const_eval/src/errors.rs

```diff
@@ -476,6 +476,150 @@ fn bad_pointer_message(msg: CheckInAllocMsg, handler: &Handler) -> String {
     handler.eagerly_translate_to_string(msg, [].into_iter())
 }
 
+pub struct UndefinedBehaviorInfoExt<'a>(UndefinedBehaviorInfo<'a>);
+
+impl IntoDiagnostic<'_> for UndefinedBehaviorInfoExt<'_> {
+    fn into_diagnostic(self, handler: &'_ Handler) -> DiagnosticBuilder<'_, ErrorGuaranteed> {
```

Comment:
when done you should remove `ReportErrorExt` and make error reporting use this impl.